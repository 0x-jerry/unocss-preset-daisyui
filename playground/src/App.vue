<script setup></script>

<template>
  <div class="p-4 font-bold text-2xl bg-light-5 mb-8">
    UnoCSS DaisyUI Preset Playground
  </div>
  <div class="flex flex-col">
    <div class="chat chat-start">
      <div class="chat-bubble chat-bubble-primary">
        What kind of nonsense is this
      </div>
    </div>
    <div class="chat chat-start">
      <div class="chat-bubble chat-bubble-secondary">
        Put me on the Council and not make me a Master!??
      </div>
    </div>
    <div class="chat chat-start">
      <div class="chat-bubble chat-bubble-accent">
        That's never been done in the history of the Jedi.
      </div>
    </div>
    <div class="chat chat-start">
      <div class="chat-bubble chat-bubble-neutral">It's insulting!</div>
    </div>
    <div class="chat chat-end">
      <div class="chat-bubble chat-bubble-info">Calm down, Anakin.</div>
    </div>
    <div class="chat chat-end">
      <div class="chat-bubble chat-bubble-success">
        You have been given a great honor.
      </div>
    </div>
    <div class="chat chat-end">
      <div class="chat-bubble chat-bubble-warning">
        To be on the Council at your age.
      </div>
    </div>
    <div class="chat chat-end">
      <div class="chat-bubble chat-bubble-error">
        It's never happened before.
      </div>
    </div>
  </div>
</template>

<style scoped></style>
