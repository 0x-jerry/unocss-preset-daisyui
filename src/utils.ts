import * as cssTree from 'css-tree'

export function getAllClassNames(ast: cssTree.CssNode): string[] {
  const classNames = new Set<string>()

  cssTree.walk(ast, {
    visit: 'ClassSelector',
    enter(node) {
      const name = node.name.replaceAll('\\/', '/')

      if (!name.includes(':')) {
        classNames.add(name)
      }
    },
  })

  return [...classNames]
}
