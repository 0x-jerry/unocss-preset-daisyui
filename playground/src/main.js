import { createApp } from 'vue'
import 'uno.css'
import 'daisyui/theme/light.css'
import App from './App.vue'

createApp(App).mount('#app')
