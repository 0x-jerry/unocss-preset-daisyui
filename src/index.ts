import { readFile } from 'node:fs/promises'
import {
  definePreset,
  type Preflight,
  type Preset,
  type PresetFactoryAwaitable,
  type Rule,
} from 'unocss'
import { getAllClassNames, globForDaisyUI, processByPostCss } from './utils'

async function calcClassContentMap() {
  const clxNameFilesMap = new Map<string, Set<string>>()
  const fileContent = new Map<string, string>()

  const files = globForDaisyUI([
    //
    'utilities/*.css',
    'colors/*.css',
    'components/*.css',
  ])

  const p = files.map(async (file) => {
    const content = await readFile(file, 'utf-8')

    const cssContent = await processByPostCss(content)

    fileContent.set(file, cssContent)

    const names = getAllClassNames(cssContent, { source: false })

    names.forEach((_, name) => {
      if (!clxNameFilesMap.has(name)) {
        clxNameFilesMap.set(name, new Set())
      }

      clxNameFilesMap.get(name)?.add(file)
    })
  })

  await Promise.all(p)

  return {
    fileMap: fileContent,
    clxNameMap: clxNameFilesMap,
  }
}

export const presetDaisyui: PresetFactoryAwaitable<object, undefined> =
  definePreset(async () => {
    const { fileMap, clxNameMap } = await calcClassContentMap()

    const autocompletes: string[] = [...clxNameMap.keys()]

    const rules: Rule[] = autocompletes.map((name) => [
      name,
      [`/* daisyui: ${name} */`],
    ])

    const extraIncludeFiles = new Set<string>()

    const preflight: Preflight = {
      getCSS(ctx) {
        const builtinFiles = globForDaisyUI(['base/*.css'])

        const includeFiles = new Set<string>(builtinFiles)

        ctx.generator.activatedRules.forEach((rule) => {
          const name = rule[0]
          if (name instanceof RegExp) {
            return
          }

          if (clxNameMap.has(name)) {
            clxNameMap.get(name)?.forEach((file) => {
              includeFiles.add(file)
            })
          }
        })

        const cssFiles = [...includeFiles, ...extraIncludeFiles]
        const css = cssFiles.map((file) => {
          const content = fileMap.get(file)
          return content || ''
        })

        return css.join('\n')
      },
    }

    const preset: Preset = {
      name: 'uno-preset-daisyui',
      rules,
      preflights: [preflight],
      transformers: [
        {
          name: 'daisyui-scanner',
          transform(_code, _id, ctx) {
            // check variant classes like `hover:xxx`, `2xl:xxx`
            for (const token of ctx.tokens) {
              if (token.includes(':')) {
                if (clxNameMap.has(token)) {
                  clxNameMap.get(token)?.forEach((file) => {
                    extraIncludeFiles.add(file)
                  })
                }
              }
            }
          },
        },
      ],
    }

    return preset
  })

export default presetDaisyui
