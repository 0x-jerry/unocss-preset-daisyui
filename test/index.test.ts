import { parse } from 'css-tree'
import { readFile } from 'node:fs/promises'
import { fileURLToPath } from 'node:url'
import { getAllClassNames } from '../src/utils'
import { createRequire } from 'node:module'

describe('utils', () => {
  it('get class names', async () => {
    const filePath = createRequire(import.meta.url).resolve(
      'daisyui/daisyui.css',
    )

    const names = getAllClassNames(parse(await readFile(filePath, 'utf-8')))
    expect(names).matchSnapshot()
  })
})
