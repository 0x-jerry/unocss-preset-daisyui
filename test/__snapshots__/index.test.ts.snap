// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`utils > get class names 1`] = `
[
  "theme-controller",
  "chat",
  "modal-open",
  "modal",
  "modal-toggle",
  "drawer",
  "drawer-open",
  "drawer-toggle",
  "modal-start",
  "modal-end",
  "textarea",
  "textarea-ghost",
  "textarea-neutral",
  "textarea-primary",
  "textarea-secondary",
  "textarea-accent",
  "textarea-info",
  "textarea-success",
  "textarea-warning",
  "textarea-error",
  "textarea-xs",
  "textarea-sm",
  "textarea-md",
  "textarea-lg",
  "textarea-xl",
  "avatar-group",
  "avatar",
  "avatar-placeholder",
  "avatar-online",
  "avatar-offline",
  "loading",
  "loading-spinner",
  "loading-dots",
  "loading-ring",
  "loading-ball",
  "loading-bars",
  "loading-infinity",
  "loading-xs",
  "loading-sm",
  "loading-md",
  "loading-lg",
  "loading-xl",
  "diff",
  "diff-item-1",
  "diff-resizer",
  "diff-item-2",
  "list",
  "list-row",
  "list-col-grow",
  "list-col-wrap",
  "kbd",
  "kbd-xs",
  "kbd-sm",
  "kbd-md",
  "kbd-lg",
  "kbd-xl",
  "navbar",
  "navbar-start",
  "navbar-center",
  "navbar-end",
  "tabs",
  "tab",
  "tab-active",
  "tab-content",
  "tab-disabled",
  "tabs-border",
  "tabs-lift",
  "tabs-top",
  "tabs-bottom",
  "tabs-box",
  "tabs-xs",
  "tabs-sm",
  "tabs-md",
  "tabs-lg",
  "tabs-xl",
  "file-input",
  "file-input-ghost",
  "file-input-neutral",
  "file-input-primary",
  "file-input-secondary",
  "file-input-accent",
  "file-input-info",
  "file-input-success",
  "file-input-warning",
  "file-input-error",
  "file-input-xs",
  "file-input-sm",
  "file-input-md",
  "file-input-lg",
  "file-input-xl",
  "input",
  "input-ghost",
  "input-neutral",
  "input-primary",
  "input-secondary",
  "input-accent",
  "input-info",
  "input-success",
  "input-warning",
  "input-error",
  "input-xs",
  "input-sm",
  "input-md",
  "input-lg",
  "input-xl",
  "table",
  "row-hover",
  "table-pin-rows",
  "table-pin-cols",
  "table-zebra",
  "table-xs",
  "table-sm",
  "table-md",
  "table-lg",
  "table-xl",
  "mockup-code",
  "mockup-window",
  "mockup-browser",
  "mockup-browser-toolbar",
  "mockup-phone",
  "mockup-phone-camera",
  "mockup-phone-display",
  "countdown",
  "label",
  "select",
  "floating-label",
  "select-xs",
  "select-sm",
  "select-md",
  "select-lg",
  "select-xl",
  "collapse",
  "collapse-close",
  "collapse-content",
  "collapse-title",
  "collapse-arrow",
  "collapse-open",
  "collapse-plus",
  "radial-progress",
  "card",
  "card-border",
  "card-dash",
  "image-full",
  "card-body",
  "card-title",
  "card-actions",
  "card-xs",
  "card-sm",
  "card-md",
  "card-lg",
  "card-xl",
  "card-side",
  "hero",
  "hero-overlay",
  "hero-content",
  "dropdown",
  "dropdown-content",
  "dropdown-open",
  "dropdown-hover",
  "dropdown-start",
  "dropdown-left",
  "dropdown-right",
  "dropdown-center",
  "dropdown-end",
  "dropdown-bottom",
  "dropdown-top",
  "skeleton",
  "radio",
  "radio-primary",
  "radio-secondary",
  "radio-accent",
  "radio-neutral",
  "radio-info",
  "radio-success",
  "radio-warning",
  "radio-error",
  "radio-xs",
  "radio-sm",
  "radio-md",
  "radio-lg",
  "radio-xl",
  "footer",
  "footer-center",
  "footer-title",
  "footer-horizontal",
  "footer-vertical",
  "tooltip",
  "tooltip-content",
  "tooltip-top",
  "tooltip-bottom",
  "tooltip-left",
  "tooltip-right",
  "tooltip-open",
  "tooltip-primary",
  "tooltip-secondary",
  "tooltip-accent",
  "tooltip-info",
  "tooltip-success",
  "tooltip-warning",
  "tooltip-error",
  "progress",
  "progress-primary",
  "progress-secondary",
  "progress-accent",
  "progress-neutral",
  "progress-info",
  "progress-success",
  "progress-warning",
  "progress-error",
  "filter",
  "filter-reset",
  "toggle",
  "toggle-primary",
  "toggle-secondary",
  "toggle-accent",
  "toggle-neutral",
  "toggle-success",
  "toggle-warning",
  "toggle-info",
  "toggle-error",
  "toggle-xs",
  "toggle-sm",
  "toggle-md",
  "toggle-lg",
  "toggle-xl",
  "btn",
  "btn-active",
  "btn-primary",
  "btn-secondary",
  "btn-accent",
  "btn-neutral",
  "btn-info",
  "btn-success",
  "btn-warning",
  "btn-error",
  "btn-ghost",
  "btn-disabled",
  "btn-link",
  "btn-outline",
  "btn-dash",
  "btn-soft",
  "btn-xs",
  "btn-sm",
  "btn-md",
  "btn-lg",
  "btn-xl",
  "btn-square",
  "btn-circle",
  "btn-wide",
  "btn-block",
  "link",
  "link-hover",
  "link-primary",
  "link-secondary",
  "link-accent",
  "link-neutral",
  "link-success",
  "link-info",
  "link-warning",
  "link-error",
  "range",
  "range-primary",
  "range-secondary",
  "range-accent",
  "range-neutral",
  "range-success",
  "range-warning",
  "range-info",
  "range-error",
  "range-xs",
  "range-sm",
  "range-md",
  "range-lg",
  "range-xl",
  "menu",
  "menu-dropdown",
  "menu-dropdown-show",
  "menu-title",
  "menu-dropdown-toggle",
  "disabled",
  "menu-active",
  "menu-focus",
  "badge",
  "menu-disabled",
  "menu-horizontal",
  "menu-vertical",
  "menu-xs",
  "menu-sm",
  "menu-md",
  "menu-lg",
  "menu-xl",
  "modal-box",
  "modal-action",
  "modal-backdrop",
  "modal-top",
  "modal-middle",
  "modal-bottom",
  "toast",
  "toast-start",
  "toast-center",
  "toast-end",
  "toast-bottom",
  "toast-middle",
  "toast-top",
  "breadcrumbs",
  "chat-bubble",
  "chat-bubble-primary",
  "chat-bubble-secondary",
  "chat-bubble-accent",
  "chat-bubble-neutral",
  "chat-bubble-info",
  "chat-bubble-success",
  "chat-bubble-warning",
  "chat-bubble-error",
  "chat-image",
  "chat-header",
  "chat-footer",
  "chat-start",
  "chat-end",
  "stats",
  "stat",
  "stat-figure",
  "stat-title",
  "stat-value",
  "stat-desc",
  "stat-actions",
  "stats-horizontal",
  "stats-vertical",
  "indicator",
  "indicator-item",
  "indicator-start",
  "indicator-center",
  "indicator-end",
  "indicator-bottom",
  "indicator-middle",
  "indicator-top",
  "divider",
  "divider-horizontal",
  "divider-vertical",
  "divider-neutral",
  "divider-primary",
  "divider-secondary",
  "divider-accent",
  "divider-success",
  "divider-warning",
  "divider-info",
  "divider-error",
  "divider-start",
  "divider-end",
  "alert",
  "alert-outline",
  "alert-dash",
  "alert-soft",
  "alert-info",
  "alert-success",
  "alert-warning",
  "alert-error",
  "alert-vertical",
  "alert-horizontal",
  "validator",
  "validator-hint",
  "steps",
  "step",
  "step-icon",
  "step-neutral",
  "step-primary",
  "step-secondary",
  "step-accent",
  "step-info",
  "step-success",
  "step-warning",
  "step-error",
  "steps-horizontal",
  "steps-vertical",
  "status",
  "status-primary",
  "status-secondary",
  "status-accent",
  "status-neutral",
  "status-info",
  "status-success",
  "status-warning",
  "status-error",
  "status-xs",
  "status-sm",
  "status-md",
  "status-lg",
  "status-xl",
  "carousel",
  "carousel-vertical",
  "carousel-horizontal",
  "carousel-item",
  "carousel-start",
  "carousel-center",
  "carousel-end",
  "mask",
  "mask-half-1",
  "mask-half-2",
  "mask-squircle",
  "mask-decagon",
  "mask-diamond",
  "mask-heart",
  "mask-hexagon",
  "mask-hexagon-2",
  "mask-circle",
  "mask-pentagon",
  "mask-star",
  "mask-star-2",
  "mask-triangle",
  "mask-triangle-2",
  "mask-triangle-3",
  "mask-triangle-4",
  "rating",
  "rating-hidden",
  "rating-xs",
  "rating-sm",
  "rating-md",
  "rating-lg",
  "rating-xl",
  "rating-half",
  "drawer-content",
  "drawer-side",
  "drawer-overlay",
  "drawer-button",
  "drawer-end",
  "dock",
  "dock-label",
  "dock-active",
  "dock-xs",
  "dock-sm",
  "dock-md",
  "dock-lg",
  "dock-xl",
  "badge-outline",
  "badge-dash",
  "badge-soft",
  "badge-primary",
  "badge-secondary",
  "badge-accent",
  "badge-neutral",
  "badge-info",
  "badge-success",
  "badge-warning",
  "badge-error",
  "badge-ghost",
  "badge-xs",
  "badge-sm",
  "badge-md",
  "badge-lg",
  "badge-xl",
  "stack",
  "stack-bottom",
  "stack-top",
  "stack-start",
  "stack-end",
  "checkbox",
  "checkbox-primary",
  "checkbox-secondary",
  "checkbox-accent",
  "checkbox-neutral",
  "checkbox-info",
  "checkbox-success",
  "checkbox-warning",
  "checkbox-error",
  "checkbox-xs",
  "checkbox-sm",
  "checkbox-md",
  "checkbox-lg",
  "checkbox-xl",
  "cally",
  "react-day-picker",
  "rdp-nav",
  "rdp-chevron",
  "rdp-day",
  "rdp-day_button",
  "rdp-caption_label",
  "rdp-button_next",
  "rdp-button_previous",
  "rdp-dropdowns",
  "rdp-dropdown",
  "rdp-dropdown_root",
  "rdp-month_caption",
  "rdp-months",
  "rdp-month_grid",
  "rdp-weekday",
  "rdp-week_number",
  "rdp-today",
  "rdp-outside",
  "rdp-selected",
  "rdp-disabled",
  "rdp-hidden",
  "rdp-range_start",
  "rdp-range_middle",
  "rdp-range_end",
  "rdp-focusable",
  "rdp-footer",
  "pika-single",
  "is-hidden",
  "is-bound",
  "pika-lendar",
  "pika-title",
  "pika-label",
  "pika-prev",
  "pika-next",
  "is-disabled",
  "pika-select",
  "pika-table",
  "pika-button",
  "pika-week",
  "is-today",
  "is-selected",
  "has-event",
  "is-inrange",
  "is-startrange",
  "is-endrange",
  "is-outside-current-month",
  "is-selection-disabled",
  "pika-row",
  "pick-whole-week",
  "swap",
  "swap-on",
  "swap-indeterminate",
  "swap-off",
  "swap-active",
  "swap-rotate",
  "swap-flip",
  "timeline",
  "timeline-middle",
  "timeline-box",
  "timeline-start",
  "timeline-end",
  "timeline-compact",
  "timeline-vertical",
  "timeline-snap-icon",
  "timeline-horizontal",
  "fieldset",
  "fieldset-legend",
  "fieldset-label",
  "select-ghost",
  "select-neutral",
  "select-primary",
  "select-secondary",
  "select-accent",
  "select-info",
  "select-success",
  "select-warning",
  "select-error",
  "join",
  "join-item",
  "join-vertical",
  "join-horizontal",
  "prose",
  "rounded-box",
  "rounded-field",
  "rounded-selector",
  "rounded-t-box",
  "rounded-b-box",
  "rounded-l-box",
  "rounded-r-box",
  "rounded-tl-box",
  "rounded-tr-box",
  "rounded-br-box",
  "rounded-bl-box",
  "rounded-t-field",
  "rounded-b-field",
  "rounded-l-field",
  "rounded-r-field",
  "rounded-tl-field",
  "rounded-tr-field",
  "rounded-br-field",
  "rounded-bl-field",
  "rounded-t-selector",
  "rounded-b-selector",
  "rounded-l-selector",
  "rounded-r-selector",
  "rounded-tl-selector",
  "rounded-tr-selector",
  "rounded-br-selector",
  "rounded-bl-selector",
  "glass",
  "bg-base-100",
  "bg-base-100/10",
  "bg-base-100/20",
  "bg-base-100/30",
  "bg-base-100/40",
  "bg-base-100/50",
  "bg-base-100/60",
  "bg-base-100/70",
  "bg-base-100/80",
  "bg-base-100/90",
  "text-base-100",
  "text-base-100/10",
  "text-base-100/20",
  "text-base-100/30",
  "text-base-100/40",
  "text-base-100/50",
  "text-base-100/60",
  "text-base-100/70",
  "text-base-100/80",
  "text-base-100/90",
  "border-base-100",
  "border-base-100/10",
  "border-base-100/20",
  "border-base-100/30",
  "border-base-100/40",
  "border-base-100/50",
  "border-base-100/60",
  "border-base-100/70",
  "border-base-100/80",
  "border-base-100/90",
  "bg-base-200",
  "bg-base-200/10",
  "bg-base-200/20",
  "bg-base-200/30",
  "bg-base-200/40",
  "bg-base-200/50",
  "bg-base-200/60",
  "bg-base-200/70",
  "bg-base-200/80",
  "bg-base-200/90",
  "text-base-200",
  "text-base-200/10",
  "text-base-200/20",
  "text-base-200/30",
  "text-base-200/40",
  "text-base-200/50",
  "text-base-200/60",
  "text-base-200/70",
  "text-base-200/80",
  "text-base-200/90",
  "border-base-200",
  "border-base-200/10",
  "border-base-200/20",
  "border-base-200/30",
  "border-base-200/40",
  "border-base-200/50",
  "border-base-200/60",
  "border-base-200/70",
  "border-base-200/80",
  "border-base-200/90",
  "bg-base-300",
  "bg-base-300/10",
  "bg-base-300/20",
  "bg-base-300/30",
  "bg-base-300/40",
  "bg-base-300/50",
  "bg-base-300/60",
  "bg-base-300/70",
  "bg-base-300/80",
  "bg-base-300/90",
  "text-base-300",
  "text-base-300/10",
  "text-base-300/20",
  "text-base-300/30",
  "text-base-300/40",
  "text-base-300/50",
  "text-base-300/60",
  "text-base-300/70",
  "text-base-300/80",
  "text-base-300/90",
  "border-base-300",
  "border-base-300/10",
  "border-base-300/20",
  "border-base-300/30",
  "border-base-300/40",
  "border-base-300/50",
  "border-base-300/60",
  "border-base-300/70",
  "border-base-300/80",
  "border-base-300/90",
  "bg-base-content",
  "bg-base-content/10",
  "bg-base-content/20",
  "bg-base-content/30",
  "bg-base-content/40",
  "bg-base-content/50",
  "bg-base-content/60",
  "bg-base-content/70",
  "bg-base-content/80",
  "bg-base-content/90",
  "text-base-content",
  "text-base-content/10",
  "text-base-content/20",
  "text-base-content/30",
  "text-base-content/40",
  "text-base-content/50",
  "text-base-content/60",
  "text-base-content/70",
  "text-base-content/80",
  "text-base-content/90",
  "border-base-content",
  "border-base-content/10",
  "border-base-content/20",
  "border-base-content/30",
  "border-base-content/40",
  "border-base-content/50",
  "border-base-content/60",
  "border-base-content/70",
  "border-base-content/80",
  "border-base-content/90",
  "bg-primary",
  "bg-primary/10",
  "bg-primary/20",
  "bg-primary/30",
  "bg-primary/40",
  "bg-primary/50",
  "bg-primary/60",
  "bg-primary/70",
  "bg-primary/80",
  "bg-primary/90",
  "text-primary",
  "text-primary/10",
  "text-primary/20",
  "text-primary/30",
  "text-primary/40",
  "text-primary/50",
  "text-primary/60",
  "text-primary/70",
  "text-primary/80",
  "text-primary/90",
  "border-primary",
  "border-primary/10",
  "border-primary/20",
  "border-primary/30",
  "border-primary/40",
  "border-primary/50",
  "border-primary/60",
  "border-primary/70",
  "border-primary/80",
  "border-primary/90",
  "bg-primary-content",
  "bg-primary-content/10",
  "bg-primary-content/20",
  "bg-primary-content/30",
  "bg-primary-content/40",
  "bg-primary-content/50",
  "bg-primary-content/60",
  "bg-primary-content/70",
  "bg-primary-content/80",
  "bg-primary-content/90",
  "text-primary-content",
  "text-primary-content/10",
  "text-primary-content/20",
  "text-primary-content/30",
  "text-primary-content/40",
  "text-primary-content/50",
  "text-primary-content/60",
  "text-primary-content/70",
  "text-primary-content/80",
  "text-primary-content/90",
  "border-primary-content",
  "border-primary-content/10",
  "border-primary-content/20",
  "border-primary-content/30",
  "border-primary-content/40",
  "border-primary-content/50",
  "border-primary-content/60",
  "border-primary-content/70",
  "border-primary-content/80",
  "border-primary-content/90",
  "bg-secondary",
  "bg-secondary/10",
  "bg-secondary/20",
  "bg-secondary/30",
  "bg-secondary/40",
  "bg-secondary/50",
  "bg-secondary/60",
  "bg-secondary/70",
  "bg-secondary/80",
  "bg-secondary/90",
  "text-secondary",
  "text-secondary/10",
  "text-secondary/20",
  "text-secondary/30",
  "text-secondary/40",
  "text-secondary/50",
  "text-secondary/60",
  "text-secondary/70",
  "text-secondary/80",
  "text-secondary/90",
  "border-secondary",
  "border-secondary/10",
  "border-secondary/20",
  "border-secondary/30",
  "border-secondary/40",
  "border-secondary/50",
  "border-secondary/60",
  "border-secondary/70",
  "border-secondary/80",
  "border-secondary/90",
  "bg-secondary-content",
  "bg-secondary-content/10",
  "bg-secondary-content/20",
  "bg-secondary-content/30",
  "bg-secondary-content/40",
  "bg-secondary-content/50",
  "bg-secondary-content/60",
  "bg-secondary-content/70",
  "bg-secondary-content/80",
  "bg-secondary-content/90",
  "text-secondary-content",
  "text-secondary-content/10",
  "text-secondary-content/20",
  "text-secondary-content/30",
  "text-secondary-content/40",
  "text-secondary-content/50",
  "text-secondary-content/60",
  "text-secondary-content/70",
  "text-secondary-content/80",
  "text-secondary-content/90",
  "border-secondary-content",
  "border-secondary-content/10",
  "border-secondary-content/20",
  "border-secondary-content/30",
  "border-secondary-content/40",
  "border-secondary-content/50",
  "border-secondary-content/60",
  "border-secondary-content/70",
  "border-secondary-content/80",
  "border-secondary-content/90",
  "bg-accent",
  "bg-accent/10",
  "bg-accent/20",
  "bg-accent/30",
  "bg-accent/40",
  "bg-accent/50",
  "bg-accent/60",
  "bg-accent/70",
  "bg-accent/80",
  "bg-accent/90",
  "text-accent",
  "text-accent/10",
  "text-accent/20",
  "text-accent/30",
  "text-accent/40",
  "text-accent/50",
  "text-accent/60",
  "text-accent/70",
  "text-accent/80",
  "text-accent/90",
  "border-accent",
  "border-accent/10",
  "border-accent/20",
  "border-accent/30",
  "border-accent/40",
  "border-accent/50",
  "border-accent/60",
  "border-accent/70",
  "border-accent/80",
  "border-accent/90",
  "bg-accent-content",
  "bg-accent-content/10",
  "bg-accent-content/20",
  "bg-accent-content/30",
  "bg-accent-content/40",
  "bg-accent-content/50",
  "bg-accent-content/60",
  "bg-accent-content/70",
  "bg-accent-content/80",
  "bg-accent-content/90",
  "text-accent-content",
  "text-accent-content/10",
  "text-accent-content/20",
  "text-accent-content/30",
  "text-accent-content/40",
  "text-accent-content/50",
  "text-accent-content/60",
  "text-accent-content/70",
  "text-accent-content/80",
  "text-accent-content/90",
  "border-accent-content",
  "border-accent-content/10",
  "border-accent-content/20",
  "border-accent-content/30",
  "border-accent-content/40",
  "border-accent-content/50",
  "border-accent-content/60",
  "border-accent-content/70",
  "border-accent-content/80",
  "border-accent-content/90",
  "bg-neutral",
  "bg-neutral/10",
  "bg-neutral/20",
  "bg-neutral/30",
  "bg-neutral/40",
  "bg-neutral/50",
  "bg-neutral/60",
  "bg-neutral/70",
  "bg-neutral/80",
  "bg-neutral/90",
  "text-neutral",
  "text-neutral/10",
  "text-neutral/20",
  "text-neutral/30",
  "text-neutral/40",
  "text-neutral/50",
  "text-neutral/60",
  "text-neutral/70",
  "text-neutral/80",
  "text-neutral/90",
  "border-neutral",
  "border-neutral/10",
  "border-neutral/20",
  "border-neutral/30",
  "border-neutral/40",
  "border-neutral/50",
  "border-neutral/60",
  "border-neutral/70",
  "border-neutral/80",
  "border-neutral/90",
  "bg-neutral-content",
  "bg-neutral-content/10",
  "bg-neutral-content/20",
  "bg-neutral-content/30",
  "bg-neutral-content/40",
  "bg-neutral-content/50",
  "bg-neutral-content/60",
  "bg-neutral-content/70",
  "bg-neutral-content/80",
  "bg-neutral-content/90",
  "text-neutral-content",
  "text-neutral-content/10",
  "text-neutral-content/20",
  "text-neutral-content/30",
  "text-neutral-content/40",
  "text-neutral-content/50",
  "text-neutral-content/60",
  "text-neutral-content/70",
  "text-neutral-content/80",
  "text-neutral-content/90",
  "border-neutral-content",
  "border-neutral-content/10",
  "border-neutral-content/20",
  "border-neutral-content/30",
  "border-neutral-content/40",
  "border-neutral-content/50",
  "border-neutral-content/60",
  "border-neutral-content/70",
  "border-neutral-content/80",
  "border-neutral-content/90",
  "bg-info",
  "bg-info/10",
  "bg-info/20",
  "bg-info/30",
  "bg-info/40",
  "bg-info/50",
  "bg-info/60",
  "bg-info/70",
  "bg-info/80",
  "bg-info/90",
  "text-info",
  "text-info/10",
  "text-info/20",
  "text-info/30",
  "text-info/40",
  "text-info/50",
  "text-info/60",
  "text-info/70",
  "text-info/80",
  "text-info/90",
  "border-info",
  "border-info/10",
  "border-info/20",
  "border-info/30",
  "border-info/40",
  "border-info/50",
  "border-info/60",
  "border-info/70",
  "border-info/80",
  "border-info/90",
  "bg-info-content",
  "bg-info-content/10",
  "bg-info-content/20",
  "bg-info-content/30",
  "bg-info-content/40",
  "bg-info-content/50",
  "bg-info-content/60",
  "bg-info-content/70",
  "bg-info-content/80",
  "bg-info-content/90",
  "text-info-content",
  "text-info-content/10",
  "text-info-content/20",
  "text-info-content/30",
  "text-info-content/40",
  "text-info-content/50",
  "text-info-content/60",
  "text-info-content/70",
  "text-info-content/80",
  "text-info-content/90",
  "border-info-content",
  "border-info-content/10",
  "border-info-content/20",
  "border-info-content/30",
  "border-info-content/40",
  "border-info-content/50",
  "border-info-content/60",
  "border-info-content/70",
  "border-info-content/80",
  "border-info-content/90",
  "bg-success",
  "bg-success/10",
  "bg-success/20",
  "bg-success/30",
  "bg-success/40",
  "bg-success/50",
  "bg-success/60",
  "bg-success/70",
  "bg-success/80",
  "bg-success/90",
  "text-success",
  "text-success/10",
  "text-success/20",
  "text-success/30",
  "text-success/40",
  "text-success/50",
  "text-success/60",
  "text-success/70",
  "text-success/80",
  "text-success/90",
  "border-success",
  "border-success/10",
  "border-success/20",
  "border-success/30",
  "border-success/40",
  "border-success/50",
  "border-success/60",
  "border-success/70",
  "border-success/80",
  "border-success/90",
  "bg-success-content",
  "bg-success-content/10",
  "bg-success-content/20",
  "bg-success-content/30",
  "bg-success-content/40",
  "bg-success-content/50",
  "bg-success-content/60",
  "bg-success-content/70",
  "bg-success-content/80",
  "bg-success-content/90",
  "text-success-content",
  "text-success-content/10",
  "text-success-content/20",
  "text-success-content/30",
  "text-success-content/40",
  "text-success-content/50",
  "text-success-content/60",
  "text-success-content/70",
  "text-success-content/80",
  "text-success-content/90",
  "border-success-content",
  "border-success-content/10",
  "border-success-content/20",
  "border-success-content/30",
  "border-success-content/40",
  "border-success-content/50",
  "border-success-content/60",
  "border-success-content/70",
  "border-success-content/80",
  "border-success-content/90",
  "bg-warning",
  "bg-warning/10",
  "bg-warning/20",
  "bg-warning/30",
  "bg-warning/40",
  "bg-warning/50",
  "bg-warning/60",
  "bg-warning/70",
  "bg-warning/80",
  "bg-warning/90",
  "text-warning",
  "text-warning/10",
  "text-warning/20",
  "text-warning/30",
  "text-warning/40",
  "text-warning/50",
  "text-warning/60",
  "text-warning/70",
  "text-warning/80",
  "text-warning/90",
  "border-warning",
  "border-warning/10",
  "border-warning/20",
  "border-warning/30",
  "border-warning/40",
  "border-warning/50",
  "border-warning/60",
  "border-warning/70",
  "border-warning/80",
  "border-warning/90",
  "bg-warning-content",
  "bg-warning-content/10",
  "bg-warning-content/20",
  "bg-warning-content/30",
  "bg-warning-content/40",
  "bg-warning-content/50",
  "bg-warning-content/60",
  "bg-warning-content/70",
  "bg-warning-content/80",
  "bg-warning-content/90",
  "text-warning-content",
  "text-warning-content/10",
  "text-warning-content/20",
  "text-warning-content/30",
  "text-warning-content/40",
  "text-warning-content/50",
  "text-warning-content/60",
  "text-warning-content/70",
  "text-warning-content/80",
  "text-warning-content/90",
  "border-warning-content",
  "border-warning-content/10",
  "border-warning-content/20",
  "border-warning-content/30",
  "border-warning-content/40",
  "border-warning-content/50",
  "border-warning-content/60",
  "border-warning-content/70",
  "border-warning-content/80",
  "border-warning-content/90",
  "bg-error",
  "bg-error/10",
  "bg-error/20",
  "bg-error/30",
  "bg-error/40",
  "bg-error/50",
  "bg-error/60",
  "bg-error/70",
  "bg-error/80",
  "bg-error/90",
  "text-error",
  "text-error/10",
  "text-error/20",
  "text-error/30",
  "text-error/40",
  "text-error/50",
  "text-error/60",
  "text-error/70",
  "text-error/80",
  "text-error/90",
  "border-error",
  "border-error/10",
  "border-error/20",
  "border-error/30",
  "border-error/40",
  "border-error/50",
  "border-error/60",
  "border-error/70",
  "border-error/80",
  "border-error/90",
  "bg-error-content",
  "bg-error-content/10",
  "bg-error-content/20",
  "bg-error-content/30",
  "bg-error-content/40",
  "bg-error-content/50",
  "bg-error-content/60",
  "bg-error-content/70",
  "bg-error-content/80",
  "bg-error-content/90",
  "text-error-content",
  "text-error-content/10",
  "text-error-content/20",
  "text-error-content/30",
  "text-error-content/40",
  "text-error-content/50",
  "text-error-content/60",
  "text-error-content/70",
  "text-error-content/80",
  "text-error-content/90",
  "border-error-content",
  "border-error-content/10",
  "border-error-content/20",
  "border-error-content/30",
  "border-error-content/40",
  "border-error-content/50",
  "border-error-content/60",
  "border-error-content/70",
  "border-error-content/80",
  "border-error-content/90",
]
`;
